lib/BridgedRoom.js

```diff
@@ -188,81 +190,172 @@ BridgedRoom.prototype.toEntry = function() {
     return entry;
 };
 
-BridgedRoom.prototype.onMatrixMessage = function(message) {
+BridgedRoom.prototype.onMatrixRedaction = async function(message) {
+    if (!this._slack_bot_token) return Promise.resolve();
+
+    const eventStore = this._main.getEventStore();
+    const event = await eventStore.getEntryByMatrixId(message.room_id, message.redacts);
+
+    // If we don't get an event then exit
+    if (event === null) {
+        log.debug("Could not find event to delete.");
+        return Promise.resolve();
+    }
+
+    var body = {channel: this._slack_channel_id,
+                ts: event.remoteEventId,
+                as_user: false};
+
+    const sendMessageParams = {
+        method: "POST",
+        json: true,
+        uri: "https://slack.com/api/chat.delete",
+        body: body,
+    };
+
+    if (this._slack_bot_token) {
```

Comment:
Do we have to do this every time? Couldn't we make a wrapper func for authenticated requests?

Reply:
:man_shrugging: